doc/design/eks-jupyterhub.md

```diff
@@ -0,0 +1,70 @@
+# Deployment of User Jupyterhubs with EKS
+
+## Current Setup
+
+Currently, we use an [Ansible
+playbook](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_intro.html)
+as the top level interface.  An admin first sets up an EC2 instance, and sets
+some Amazon Web Services configuration, and then runs the playbook
+against the EC2 instance. The playbook then creates all remaining Amazon
+resources, uses [`kops`](https://kops.sigs.k8s.io/) to install and
+configure Kubernetes, and then installs the official [Jupyterhub Helm
+Chart](https://hub.jupyter.org/helm-chart/). DANDI users authenticate
+with GitHub OAuth, and can create their own user-space `jupyterhub`
+instances.
+
+### Limitations
+
+ - This approach cannot yet be fully automated, and requires administrator configuration in the AWS console.
+ - The Ansible playbook is based on the [zero to jupyterhub](https://github.com/spacetelescope/z2jh-aws-ansible)
+   approach used by the space-telescopes group, but this approach is no longer used upstream, and
+   we currently have to maintain the whole Ansible playbook alone.
+ - The majority of the complexity of this approach is the deployment and configuration of Kubernetes itself.
+
+## Desired features
+1. Spot pricing
+1. Connection to DANDI for OAuth
+1. Custom repositories - [GESIS](https://2i2c.org/blog/2022/gesis-2i2c-collaboration-update/)
+1. Regular clean up of shared space
+1. Persistent home directories with quotas
```

Comment:
```suggestion
## Desired features

* Top priority

1. Environment/Compute
  1. Single hub with multiple groups that get allocated different resources*
  1. Bring/build your own image*
  1. Run other services on the Kubernetes layer that JupyterHub users can use
1. Costs
  1. Spot pricing (and more generally, minimize costs using any cloud feature)*
  1. Separate billing for each group
  1. Monitor access, storage, compute for each user and group
1. Storage
  1. Storage quotas*
  1. Persistent home directories
  1. Persistent shared directories
  1. Regular clean up of shared directories
1. Authentication
  1. Connection to DANDI for OAuth

```

Reply:
quota controllable by user/group

Reply:
Thank you.  I have updated my original comment.